flocker/node/agents/testtools.py

```diff
@@ -58,3 +96,156 @@ def setUp(self):
             self.client = client_factory(test_case=self)
 
     return Tests
+
+
+def blockdevice_manager_for_test(test_case):
+    """
+    Creates a blockdevice_manager that cleans itself up during test cleanup.
+
+    Cleanup is defined as unmounting all bind mounts, tmpfs mounts, and
+    blockdevice mounts.
+
+    :param test_case: The :class:`TestCase` to use to add cleanup callbacks.
+    """
+    manager = CleanupBlockDeviceManager(BlockDeviceManager())
+    test_case.addCleanup(manager.cleanup)
+    return manager
```

Comment:
:question: If you made `CleanupBlockDeviceManager` subclass `Fixture`, then this function wouldn't be necessary, as tests could just do:

```python
manager = self.useFixture(CleanupBlockDeviceManager(BlockDeviceManager())
```

But, tbh, not convinced it's worth it.